### PR TITLE
Add consul-dataplane formula

### PR DESCRIPTION
**Pull request by boxofrad**
Adds a formula for consul-dataplane. Should not be merged until the release artifacts are ready (note: the formula is missing checksums so will need to be regenerated).

Also adds support for "recommended" dependencies, as consul-dataplane requires Envoy but users may choose to build it themselves.

### CODE REVIEW

**Comment by mdeggies** [comment]
Adding @alvin-huang as a reviewer, who may have more familiarity in this repo!

**Review by modrake — APPROVED** [review]
This looks like the only case of us using depends_on in our tap, we may want to be sure to keep a close eye on the first run of this to make sure the envoy dep resolves smoothly.

File: Formula/consul-dataplane.rb [hunk 1]
```diff
@@ -0,0 +1,42 @@
+class ConsulDataplane < Formula
+  desc "Consul Dataplane"
+  homepage "https://github.com/hashicorp/consul-dataplane"
+  version "1.0.0-beta1"
+
+  if OS.mac? && Hardware::CPU.intel?
+    url "https://releases.hashicorp.com/consul-dataplane/1.0.0-beta1/consul-dataplane_1.0.0-beta1_darwin_amd64.zip"
+    sha256 ""
```
Comment (pglass) [inline Formula/consul-dataplane.rb:8]: I'll fill these in

**Review by pglass — CHANGES_REQUESTED** [review]

**Review by pglass — APPROVED** [review]

**Comment by mdeggies** [comment]
This looks good to me. I'll merge and deploy the lambda shortly

**Pull request merged**